[sidebar] Add tooltips to sidebar menu links

guilhermemntt opened the pull request:
Closes https://github.com/decred/decrediton/issues/2584.

alexlyp commented:
Needs a rebase

alexlyp commented:
Any update to tests @guilhermemntt ? still waiting on those fixes before we can merge this

guilhermemntt commented:
@alexlyp I've fixed the unit tests. Please, do not merge this PR until https://github.com/decred/pi-ui/issues/295 is fixed. It's a bug in ``pi-ui`` tooltip component that seems to affect not only this specific PR, but all others related with ``pi-ui`` tooltip. Thanks!

alexlyp commented:
OK thanks the update.  Waiting.

victorgcramos reviewed (changes requested):
Hey @guilhermemntt thanks for the PR and sorry for the late feedback.
One thing is that Tooltips alignment should be improved when sidebar is collapsed on the left. Maybe place them on the right, what do you think?

<img width="612" alt="Screen Shot 2020-10-20 at 12 09 06 PM" src="https://user-images.githubusercontent.com/22639213/96606233-6f4f5d80-12cd-11eb-9049-ff320da8f894.png">

Left some inline comments as well!

guilhermemntt commented:
Hey @victorgcramos, thanks for the review! I've just updated this PR with some fixes and improvements.

- The tooltip has been slightly moved to right, as you've sugested.

![image](https://user-images.githubusercontent.com/39631429/97125899-10964380-1714-11eb-9ceb-c9b400c27a18.png)

- I replaced the two ``map``s with a new functional component to increase the readability and modularity of the code.

@alexlyp @vctt94 sorry for the late reply. As the doubts regarding the tooltip visibility in small screens have been solved at https://github.com/decred/pi-ui/issues/295, I think this PR can be merged after the approve of the review.

guilhermemntt commented on app/components/SideBar/MenuLinks/MenuLink/MenuLink.jsx:
```diff
@@ -1,25 +1,66 @@
+  <>
+    {notifProp ? (
+      <span className={style.menuLinkNotificationIcon} data-testid="menu-link-notification-icon"></span>
+    ) : null}
```
Fixed.

guilhermemntt commented on app/components/SideBar/SideBar.module.css:
```diff
@@ -29,7 +34,13 @@
+
+.sidebar.sidebarReduced .sidebarMain,
+.sidebar.sidebarReduced .sidebarScroll {
+  overflow: visible;
```
By adding this ``CSS`` property at these components, we guarantee that their contents (tooltips here) won't be clipped by them. Otherwise, ``overflow: hidden`` would not be overrided and the we'd have this result:

![image](https://user-images.githubusercontent.com/39631429/97127296-01b19000-1718-11eb-8803-153d5655c461.png)

victorgcramos replied:
hmmmm nice, mate! Thanks for explaining :)

guilhermemntt commented on app/components/SideBar/MenuLinks/MenuLinks.jsx:
```diff
@@ -4,13 +4,32 @@ import MenuLink from "./MenuLink/MenuLink";
+const MenuList = React.memo(
+  ({ sidebarOnBottom, nodes, menuLinks, expandSideBar }) =>
+    sidebarOnBottom
+      ? menuLinks.map((menuLinkRow, index) => (
```
Fixed.

guilhermemntt commented on app/components/SideBar/MenuLinks/MenuLinks.jsx:
```diff
@@ -4,13 +4,32 @@ import MenuLink from "./MenuLink/MenuLink";
+            {menuLinkRow.map((menuLink) => {
+              const { path, link, icon, notifProp, ariaLabel } = menuLink;
+              return (
+                  <MenuLink
```
Fixed.

vctt94 reviewed (approved):
LGTM
